### PR TITLE
[branch/v16] Invoke signing function for Windows Connect binary

### PR DESCRIPTION
**Pull request by fheinecke**
Backport #41963

changelog: Fixed Teleport Connect binaries not being signed correctly.

### CODE REVIEW

File: build.assets/windows/build.ps1 [hunk 1]
```diff
@@ -473,7 +490,7 @@ function Write-Version-Objects {
     & $GoWinres simply --no-suffix --arch amd64 `
         --file-description "Teleport tsh command-line client" `
         --original-filename tsh.exe `
-        --copyright "Copyright (C) $Year Gravitational Inc." `
+        --copyright "Copyright (C) $Year Gravitational, Inc." `
```
Comment (zmb3) [inline build.assets/windows/build.ps1:493]: Best to run a tag build and then right click and go to properties on `tsh.exe` in a Windows environment. This version metadata is very picky and I seem to recall something about the comma breaking things.

File: build.assets/windows/build.ps1 [hunk 2]
```diff
@@ -484,23 +501,12 @@ function Write-Version-Objects {
...
         --icon "$TeleportSourceDirectory\e\windowsauth\installer\teleport.ico" `
         --product-name Teleport `
         --product-version $TeleportVersion `
         --file-version $TeleportVersion `
         --out "$TeleportSourceDirectory\tool\tctl\resource.syso"
-
-    # generate windowsauth version info (note the --admin flag, as the installer must run as admin)
-    & $GoWinres simply --no-suffix --arch amd64 --admin `
```
Comment (zmb3) [inline build.assets/windows/build.ps1:495]: Why was this removed?! This is necessary in order for the installer to properly request admin permissions.

**Review by zmb3 — CHANGES_REQUESTED** [review]
I'm not convinced this doesn't break things, can we verify the version metadata is correct before merging this?

**Comment by fheinecke** [comment]
@zmb3 I screwed up my local backport commands and this should now be fixed, with a significantly smaller set of changes.

**Comment by zmb3** [comment]
Looks better but I'm now confused how this fixes anything. All I can see is that `SignedBinaryPath` is no longer mandatory. Is that correct?

**Comment by fheinecke** [comment]
Yep that's pretty much it. Somehow when I did several other backports I managed to include only part of the original PR and it's breaking things due to this attribute.

**Comment by fheinecke** [comment]
Having `mandatory` set makes the field mandatory (as the name implies), and we're calling this commandlet in some places without this parameter now.

**Review by zmb3 — APPROVED** [review]

**Review by gzdunek — APPROVED** [review]

**Pull request merged**